### ISSUE TITLE
Reflected XSS via `back` parameter

### ISSUE THREAD

**Issue by vavkamil** [pirati-web/socialnisystem.cz#1]
Hi, there is a reflected Cross-Site Scripting vulnerability via `back` parameter.
XSS payload is executed after clicking on the `« Zpět` button, example:
https://socialnisystem.pirati.cz/davky/8-odmena-pestouna/?back=javascript:alert(1)

Relevant code:
https://github.com/pirati-web/socialnisystem.cz/blob/5724c82ce3177bc34b8f300256a588f983346e0d/socialsystem/core/views.py#L68

I will suggest using something like Bleach:
https://bleach.readthedocs.io/en/latest/
to prevent such issues in the future :)

**Comment by xaralis** [comment]
Thanks for pointing out, I will fix this ASAP.

**Comment by xaralis** [comment]
@vavkamil 
I've added some additional checks in 1bd25d9 which I hope are sufficient, i.e. we only allow relative URLs to be passed in `?back`. Can you please have a look?

**Comment by vavkamil** [comment]
Hey, it was a very minor issue, so thanks for the very quick response! Checking for the empty scheme is efficient in case of XSS here, but it still allows open redirect:

```
from urllib.parse import urlparse

url = '//google.com'
parsed = urlparse(url)
print(parsed)

# ParseResult(scheme='', netloc='google.com', path='', params='', query='', fragment='')
```

So something like this will still work:
https://socialnisystem.pirati.cz/davky/8-odmena-pestouna/?back=//google.com
https://socialnisystem.pirati.cz/davky/8-odmena-pestouna/?back=\\google.com

But at this point I think that it's not worth the effort to fix it.

**Comment by xaralis** [comment]
OK, I'll add check that `netloc` is empty, too :) This should fix it completely. I will release that ASAP.

**Comment by xaralis** [comment]
Fixed in fa843640553f47385425f585b044c13a968ba00f.
